bin/ingestion.js

```diff
@@ -225,6 +206,64 @@ function updateProcessors(zkClient, bootstrapList) {
     });
 }
 
+/**
+ * Refresh processors by checking zookeeper states and use the state(s) to,
+ * update ingestionPopulator's paused locations list.
+ * @param {node-zookeeper-client.Client} zkClient - zookeeper client
+ * @param {Logger.newRequestLogger} log - request logger object
+ * @param {Function} cb - callback(error)
+ * @return {undefined}
+ */
+function refreshProcessors(zkClient, log, cb) {
+    const bootstrapList = config.getBootstrapList();
+    const active = Object.keys(configuredLocations);
+    const update = bootstrapList.map(i => i.site);
+    const locations = new Set(active.concat(update));
+    const ingestionZkPath = getIngestionZkPath();
+
+    return async.each(locations, (location, next) => {
+        const path = `${ingestionZkPath}/${location}`;
+        return zkClient.getData(path, (err, data) => {
+            if (err) {
+                log.fatal('could not check location status in zookeeper',
+                    {
+                        method: 'bin::ingestion.refreshProcessors',
+                        zookeeperPath: path,
+                        error: err.message,
+                    });
+                return next(err);
+            }
+            let locationData;
+            try {
+                locationData = JSON.parse(data.toString());
+            } catch (e) {
+                log.fatal('error getting state for location', {
+                    method: 'bin::ingestion.refreshProcessors',
+                    location,
+                    error: reshapeExceptionError(e),
+                });
+                return next(e);
+            }
+            if (locationData[RESUME_NODE]) {
+                return checkAndApplyScheduleResume(zkClient, locationData,
+                    location, next);
+            }
+            if (locationData.paused) {
```

Comment:
I might be wrong, but I'm under the impression that to be complete, we should also call `ingestionPopulator.removePausedLocationState()` if the location is not paused.